base/strings/search.jl

```diff
@@ -212,20 +206,29 @@ julia> rsearch("aaabbb","b")
 6:6
 ```
 """
-function rsearch(s::AbstractString, c::Chars, i::Integer)
-    e = endof(s)
-    j = search(RevString(s), c, e-i+1)
-    j == 0 && return 0
-    e-j+1
+function rsearch(s::AbstractString, c::Chars, i::Integer=start(s))
+    if isempty(c)
+        return 1 <= i <= nextind(s, endof(s)) ? i : throw(BoundsError(s, i))
+    end
+    if i < 1 || i > nextind(s, endof(s))
+        throw(BoundsError(s, i))
+    end
+    j = i
+    while j > 0
+        d = s[j]
+        d in c && return j
+        j = prevind(s, j)
+    end
+    return 0
 end
 
 function _rsearchindex(s, t, i)
     if isempty(t)
         return 1 <= i <= nextind(s,endof(s)) ? i :
                throw(BoundsError(s, i))
     end
-    t = RevString(t)
-    rs = RevString(s)
+    t = reverse(t)
```

Comment:
Why not implement the reversed algorithm just like above?

Reply:
Wasn't sure how best to do it. Suggestions welcome.

Reply:
I don't know... Doesn't changing every `next` into `prevind` work like you did above? I admit that method looks a bit scary. Or just keep `RevString` as an internal helper type just for that method. ;-)